### DISCUSSION TITLE
3GPP CMP enrollment fails with 3.0.15 (successful with 3.0.7)

### DISCUSSION THREAD

**Discussion by henrikjehgmti** [openssl/openssl#25586] [Q&A] [answered]
Hi all

I am doing enrollment of certificates in a 3GPP operator network, and with the latest 3.0.15 release I can no longer accept the presented certificate. 

After a few hours of investigation, I came across that I needed to set the option OSSL_CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR (to 1) in order for it to work. 

Is there a way where I can get openssl cmp to actively set the OSSL_CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR without making the code change in apps/cmp/cmp,c:setup_verification_ctx()?

**Comment by t8m** [comment]
Perhaps @DDvO might be able to answer this?

**Comment by DDvO** [comment, marked as answer]
Thank you @henrikjehgmti for your question - and very sorry that I overlooked for so long.

As already explained in https://github.com/openssl/openssl/issues/23706#issuecomment-1988240652 and very recently in https://github.com/openssl/openssl/issues/27888#issuecomment-3009917861, we do not want to generally promote that special 3GPP quirk because it is inherently dangerous and can easily lead to insecurely trusting a wrong trust anchor (root) certificate.

**Comment by DDvO** [comment]
BTW, at some point there was a regression with `OSSL_CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR` that got fixed by #23814.
So at API level you should be able to use that option as documented.

**Comment by DDvO** [comment]
Update: #28015 is going to make this quirk option available from the CLI,
but not by default: only if `OPENSSL_CMP_APP_ALLOW_UNSAFE` is defined at compile time.
